qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -89,9 +96,20 @@ class HighLevelSynthesis(TransformationPass):
     ``default`` methods for all other high-level objects, including ``op_a``-objects.
     """
 
-    def __init__(self, hls_config=None):
+    def __init__(self, coupling_map: CouplingMap = None, hls_config=None):
```

Comment:
For backwards compatibility I would reverse these arguments. Python will let you specify keyword arguments positionally by default so putting `coupling_map` first will break users doing `HighLevelSynthesis(HLSConfig(...))` which was valid way to instantiate the pass before.